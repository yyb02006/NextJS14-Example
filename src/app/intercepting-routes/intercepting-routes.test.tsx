import { render, screen } from '@testing-library/react'
import InterceptingRoutesLayout from './layout'

describe('Layout Component Rendering Test', () => {
  const navButtons = [
    { label: 'to Feed', path: '/intercepting-routes/feed' },
    { label: 'to Photo', path: '/intercepting-routes/photo' },
  ]
  it(`'Global Nav in intercepting-routes Layout' 텍스트를 가진 div 요소가 있어야 한다`, () => {
    render(<InterceptingRoutesLayout children={<>children</>} />)
    const divElement = screen.getByText('Global Nav in intercepting-routes Layout', {
      selector: 'div',
    })
    expect(divElement).toBeInTheDocument()
  })
  it(`navButtons 객체의 프로퍼티로 만들어진 버튼들이 있어야 한다`, () => {
    render(<InterceptingRoutesLayout children={<>children</>} />)
    navButtons.forEach(({ label: name, path }) => {
      const link = screen.getByRole('link', { name })
      expect(link).toBeInTheDocument()
      expect(link).toHaveAttribute('href', path)
    })
  })
  it(`전체 레이아웃이 예상대로 렌더링 되어야 한다`, () => {
    const { container } = render(<InterceptingRoutesLayout children={<>children</>} />)
    expect(container).toMatchSnapshot()
  })
})
