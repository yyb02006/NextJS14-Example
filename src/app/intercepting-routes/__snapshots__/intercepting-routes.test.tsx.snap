// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component Rendering Test 전체 레이아웃이 예상대로 렌더링 되어야 한다 1`] = `
<div>
  <section>
    <nav
      class="border-b px-10"
    >
      <article
        class="flex h-16 items-center justify-around border-x"
      >
        <div
          class="flex h-full w-full flex-shrink-[2] items-center justify-center hover:bg-sky-600"
        >
          Global Nav in intercepting-routes Layout
        </div>
        <ul
          class="flex h-full w-full max-w-[70%] items-center justify-between"
        >
          <li
            class=" h-full w-full border-l"
          >
            <a
              class="flex h-full w-full items-center justify-center hover:bg-amber-500"
              href="/intercepting-routes/feed"
            >
              to Feed
            </a>
          </li>
          <li
            class=" h-full w-full border-l"
          >
            <a
              class="flex h-full w-full items-center justify-center hover:bg-amber-500"
              href="/intercepting-routes/photo"
            >
              to Photo
            </a>
          </li>
        </ul>
      </article>
    </nav>
    children
  </section>
</div>
`;
